xbmc/addons/binary-addons/AddonDll.cpp

```diff
@@ -229,17 +229,18 @@ void CAddonDll::Destroy()
     CLog::Log(LOGINFO, "ADDON: Dll Destroyed - {}", Name());
   }
 
-  ResetSettings();
+  ResetSettings(KODI::ADDONS::ADDON_SETTINGS_ID);
 
   m_initialized = false;
 }
 
-ADDON_STATUS CAddonDll::CreateInstance(ADDON_TYPE instanceType,
-                                       ADDON_INSTANCE_HANDLER instanceClass,
-                                       const std::string& instanceID,
-                                       KODI_HANDLE instance,
-                                       KODI_HANDLE parentInstance)
+ADDON_STATUS CAddonDll::CreateInstance(KODI_ADDON_INSTANCE_STRUCT* instance)
 {
+  /* Check instance structures are available, normally should never NULL, but for
+   * safeness checked here. */
```

Comment:
TBH, if those values must not be NULL, I would `assert()` here, to find the programming errors early and the hard way (via a crash with the respective crash logs for easy debugging).

Reply:
Yes is better and `assert()` becomes added.